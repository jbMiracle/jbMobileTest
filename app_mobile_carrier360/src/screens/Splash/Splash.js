import React, { Component } from 'react';
import { connect } from 'react-redux';
import { View, Image, Animated, Easing, StatusBar } from 'react-native';
import NavigationBar from 'react-native-navbar-color';
import { connectStyle } from 'native-base';
import PropTypes from 'prop-types';
import styleDef from './Splash.style';
import { App as AppActions } from '../../state/actions';
import C360Colors from '../../themes/360Colors';

const jbh36Image = require('../../assets/images/36_JB_Logo.png');
const jbhWhiteCircle = require('../../assets/images/WhiteCircle.png');

const mapStateToProps = () => ({});
const mapDispatchToProps = dispatch => ({
  genesis: () => dispatch(AppActions.genesis()),
});

class Splash extends Component {
  constructor() {
    super();
    NavigationBar.setColor(C360Colors.Black);
    StatusBar.setBarStyle('light-content');
    this.RotateValueHolder = new Animated.Value(0);
  }

  componentDidMount() {
    this.props.genesis();
    this.startImageRotation();
  }

  startImageRotation() {
    this.RotateValueHolder.setValue(0);

    Animated.timing(this.RotateValueHolder, {
      toValue: 1,
      duration: 3000,
      easing: Easing.linear,
    }).start(() => this.startImageRotation());
  }

  render() {
    const { style } = this.props;
    const rotateImage = this.RotateValueHolder.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '360deg'],
    });
    const rotatingImageStyle = {
      width: 60,
      height: 60,
      position: 'absolute',
      top: 52,
      right: 10,
      transform: [{ rotate: rotateImage }],
    };

    return (
      <View style={ style.container }>
        <Animated.View style={ { position: 'relative' } } animation='pulse'>
          <Image source={ jbh36Image } />
          <Animated.Image
            style={ rotatingImageStyle }
            source={ jbhWhiteCircle }
          />
        </Animated.View>
      </View>
    );
  }
}

Splash.propTypes = {
  style: PropTypes.object.isRequired,
  genesis: PropTypes.func.isRequired,
};

const ReduxConnectedSplash = connect(mapStateToProps, mapDispatchToProps)(Splash);
export default connectStyle('C360.Splash', styleDef)(ReduxConnectedSplash);
