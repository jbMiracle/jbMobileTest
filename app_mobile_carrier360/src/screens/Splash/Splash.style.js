export default {
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
    backgroundColor: '#007DBA',
  },
};
