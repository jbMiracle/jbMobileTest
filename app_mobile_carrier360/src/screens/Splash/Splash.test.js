import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Splash from './Splash';

it('can render an empty component', () => {
  const renderer = new ShallowRenderer();

  const props = {
    navigation: {
      navigate: () => {},
    },
  };

  renderer.render(<Splash { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
