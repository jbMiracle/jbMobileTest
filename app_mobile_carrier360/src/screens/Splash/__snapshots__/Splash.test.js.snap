// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render an empty component 1`] = `
<Connect(Splash)
  navigation={
    Object {
      "navigate": [Function],
    }
  }
  style={
    Object {
      "container": Object {
        "alignItems": "center",
        "backgroundColor": "#007DBA",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
    }
  }
  virtual={undefined}
/>
`;
